sdk/servicebus/Microsoft.Azure.ServiceBus/src/Message.cs

```diff
@@ -140,6 +146,7 @@ public string SessionId
             {
                 Message.ValidateSessionId(nameof(this.SessionId), value);
                 this.sessionId = value;
+                this.PartitionKey = value;
```

Comment:
Why not just do the same validation here as opposed to overriding the PartitionKey. It doesn't feel right to silently overwrite it when we throw for the other scenario.

Reply:
That's how other service bus SDKs do it. it was missing from .net core SDK and I am just adding that missing functionality. SDK treats SessionId as the dominant property and dominant property overwrites the other property.

---
In reply to: [465416026](https://github.com/Azure/azure-sdk-for-net/pull/13961#discussion_r465416026) [](ancestors = 465416026)